code/modules/shuttle/mini_dropship.dm

```diff
@@ -318,3 +367,214 @@
+		stack_trace("No dropship found in crash(), HOW?")
+		return
+
+	dropship.callTime = time_to_crash
+	SSshuttle.moveShuttleToDock(shuttleId, pick(SSshuttle.crash_targets), TRUE)
+
+///Handle summoning the dropship to the summoner's location; is done on the console because the process is way more complex otherwise
+/obj/machinery/computer/camera_advanced/shuttle_docker/flyable/dropship/proc/summon(atom/summoner)
+	SStgui.close_uis(src)
+	dropship.set_hijack_state(HIJACK_STATE_CALLED_DOWN)
+	playsound(summoner, SFX_ALIEN_ROAR, 50)
+	playsound(dropship.loc,'sound/effects/alert.ogg', 50, sound_range = 15)	//Use docking port loc instead of the console
+	dropship.start_hijack_timer()
+	create_landing_port()
+	my_port.forceMove(get_turf(summoner))
+	my_port.setDir(summoner.dir)
+
+	var/time_until_takeoff = dropship.timeLeft(1)
+	priority_announce(
+		"Dropship compromised by foreign actor. Interference preventing remote control. \
+		Location: [get_area_name(summoner, TRUE)]. [time_until_takeoff ? "Take-off in: [time_until_takeoff / 10] SECONDS" : "TAKE-OFF IMMINENT"].",
+		"Dropship Hijack Alert",
+		type = ANNOUNCEMENT_PRIORITY,
+		color_override = "red")
+
+	if(time_until_takeoff)
+		addtimer(CALLBACK(src, PROC_REF(summon_towards)), time_until_takeoff)
+		return
+
+	summon_towards()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/flyable/dropship/proc/summon_towards()
+	if(!dropship)
+		stack_trace("No dropship found in summon_towards(), HOW?")
+		return
+
+	//Have to account for the shuttle being either already in flight, in the main ship hangar, or already on the ground somewhere
+	//Also take into account the next_fly_state because this will be called again before the shuttle actually takes off
+	if((next_fly_state == SHUTTLE_ON_SHIP || next_fly_state == SHUTTLE_ON_GROUND) && (fly_state == SHUTTLE_ON_SHIP || fly_state == SHUTTLE_ON_GROUND))
+		take_off(TRUE)
+		//Calculate the time it takes to travel to the atmosphere; add an extra second to make sure it reaches the atmosphere
+		var/time_until_takeoff = dropship.timeLeft(1) + dropship.prearrivalTime + dropship.rechargeTime + 1 SECONDS
+		if(time_until_takeoff)
+			addtimer(CALLBACK(src, PROC_REF(summon_towards)), time_until_takeoff)
+		else
+			summon_towards()
+		return
+
+	open_prompt = FALSE	//Set this here instead of in summon() because shuttle_arrived() will set this to TRUE if take_off() was called
+	dropship.callTime = SHUTTLE_LANDING_CALLTIME
+	next_fly_state = SHUTTLE_ON_GROUND
+	dropship.set_mode(SHUTTLE_CALL)
+	SSshuttle.moveShuttleToDock(shuttleId, my_port, TRUE)
+
+/obj/item/shuttle_hijacker
+	name = "hijacker"
+	icon_state = "coin"
+	var/mob/living/carbon/xenomorph/spawner = /mob/living/carbon/xenomorph/shrike
+	var/mob/living/carbon/xenomorph/beno
+
+/obj/item/shuttle_hijacker/attack_self(mob/user)
+	beno = new spawner(locate(20, 20, 2))
+	addtimer(CALLBACK(src, PROC_REF(stuff)), 10 SECONDS)
+
+/obj/item/shuttle_hijacker/proc/stuff()
+	beno.face_atom(get_step_rand(beno))
```

Comment:
wtf?

Reply:
Testing tool I used for simulating a ruler calling down the Alamo.